File: backend/task-service/src/main/java/org/driventask/task/Payload/Request/TaskRequest.java
package org.driventask.task.Payload.Request;

import java.util.Set;

import org.driventask.task.Enum.EPriority;
import org.driventask.task.Enum.EStatus;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record TaskRequest(
    
    @NotBlank
    @NotNull
    String title,
    
    String description,
    
    EPriority priority,

    EStatus status,

    String userAssigned,

    @NotNull
    @NotBlank
    String projectId,

    Set<String> filesIncluded
) {
    
}

File: frontend2/src/app/models/task-request.ts
import { EPriority } from "./priority";
import { EStatus } from "./status";

export interface TaskRequest {
    title: String;
    description: String;
    priority: EPriority;
    status: EStatus;
    userAssigned: String;
    projectId: String;
    filesIncluded: String[];
}
